community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/RecordFormats.java

```diff
@@ -58,5 +68,22 @@ public Factory( String key, String... altKeys )
 
     RecordFormat<RelationshipTypeTokenRecord> relationshipTypeToken();
 
-    RecordFormat<RelationshipGroupRecord> relationshipGroup();
+    RecordFormat<DynamicRecord> dynamic();
+
+    boolean hasStore( StoreType store );
```

Comment:
Only the format can have that information, who else would know that?

RecordFormats and store version is conflated (as in the rest of the code historically as well). At some point we might break those two apart, but I don't think it's necessary to do that right now.